src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -50,9 +50,18 @@
     </PropertyGroup>
 
     <PropertyGroup>
-      <LangVersion Condition="'$(LangVersion)' == '' AND
-        (('$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(TargetFrameworkVersion)' == 'v3.0') OR
-         ('$(TargetFrameworkIdentifier)' == '.NETStandard' AND '$(TargetFrameworkVersion)' == 'v2.1'))">preview</LangVersion>
+      <!-- To aid certain IDE features we make a distinction between the maximum supported language version, and the version 
+           actually used during compilation. .NETCore 3.0 supports up to C# 8.0, while all others support up to 7.3 by default.
+           Note that other frameworks (e.g., Xamarin, Unity, etc.) are expected to override this by setting
+           MaxSupportedLangVersion themselves. -->
+      <!-- Strip the 'v' off the beginning of TargetFrameworkVersion so we can actually compare it as a version. -->
+      <_TargetFrameworkVersionAsVersion>$(TargetFrameworkVersion.TrimStart(vV))</_TargetFrameworkVersionAsVersion>
+      <MaxSupportedLangVersion Condition="'$(MaxSupportedLangVersion)' == '' AND
+                                          '$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND
+                                          '$(_TargetFrameworkVersionAsVersion)' >= '3.0'">8.0</MaxSupportedLangVersion>
```

Comment:
Is there any impact here of this PR implicitly changing the default langversion from "preview" to 8.0?